actors/miner/src/lib.rs

```diff
@@ -3122,29 +3135,40 @@ impl Actor {
                 // Verify unlocked funds cover both InitialPledgeRequirement and FeeDebt
                 // and repay fee debt now.
                 let fee_to_burn = repay_debts_or_abort(rt, state)?;
-
-                Ok((info, newly_vested, fee_to_burn, available_balance, state.clone()))
+                let mut amount_withdrawn = std::cmp::min(&available_balance, &params.amount_requested);
+                if amount_withdrawn.is_negative() {
+                        return Err(actor_error!(
+                            illegal_state,
+                            "negative amount to withdraw: {}",
+                            amount_withdrawn
+                        ));
+                }
+                if info.beneficiary != info.owner {
+                    let remaining_quota = info.beneficiary_term.available(rt.curr_epoch());
+                    if !remaining_quota.is_positive() {
```

Comment:
@anorth @steven004 should  allow zero for beneficiary withdraw (not exit with error), ,because in old withdraw  allow zero to pass, should flow old behavior?

Reply:
Yes I agree.

Reply:
https://github.com/filecoin-project/builtin-actors/pull/496/commits/2cdaaf647263f7766085cd050303f6942ad72510